src/compiler/checker.ts

```diff
@@ -14400,12 +14432,126 @@ namespace ts {
             }
         }
 
+        function checkUnusedLocals(node: FunctionDeclaration | MethodDeclaration | ConstructorDeclaration | FunctionExpression | ArrowFunction | ForInStatement | Block): void {

```

Comment:
nit. can we change the names of these functions:

checkUnusedLocals -> checkUnusedIdentifiers
checkUnusedIdentifiers -> checkUnusedLocals 

this way, the top-level function is checkUnusedIdentifiers, and we call checkUnusedLocals (local variables/local declarations) and checkeUnusedParamters (for function parameters)